backend/wallets/repository/wallet.py

```diff
@@ -0,0 +1,136 @@
+from django.db.models import QuerySet
+from django.core.exceptions import FieldDoesNotExist
+
+from django.contrib.auth.models import User
+
+from wallets.models import Wallet
+
+class WalletRepository:
+    """
+    Repository for the Wallet model.
+    
+    This repository contains the methods that should be used to perform actions on the Wallet model.
+
+    Methods:
+        create_wallet: A method that creates a wallet instance from the request data.
+        add_co_owners_to_wallet: A method that adds co-owners to the wallet.
+        erase_co_owners_from_wallet: A method that erases co-owners from the wallet.
+        get_all_wallets: A method that lists all wallets.
+        get_single_wallet: A method that gets the details of a wallet.
+        save_wallet: A method that saves the wallet instance.
+        delete_wallet: A method that deletes the wallet instance.
+    """
+
+    @staticmethod
+    def create_wallet(request_data: dict, owner: User) -> Wallet:
+        """
+        Create a wallet instance from the request data.
+
+        Args:
+            request_data (dict): The request data.
+            owner (User): The owner of the wallet.
+
+        Returns:
+            Wallet: The wallet instance.
+        """
+        try:
+            wallet = Wallet.objects.create(owner=owner, **request_data)            
+
+        except Exception as e:
+            raise e
```

Comment:
Same here, avoid using exceptions for flow control. If it fails you need to know it failed.